### PR TITLE
feat(opentrons-ai-client): Accordion

### PR DESCRIPTION
**Pull request by fbelginetw**
# Overview

This PR adds an Accordion component that will be consumed by the Create Protocol flow in the new AI Client.

image

## Test Plan and Hands on Testing

- Basic scenarios manually tested
- Accessibility
- Unit tests

## Changelog

- Add Opentrons AI Client Accordion component

## Review requests

- Validate if this approach for creating components fits the project structure and best practices,

## Risk assessment

- No risk

### CODE REVIEW

File: opentrons-ai-client/src/molecules/Accordion/index.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  height: auto;
+  padding: 24px 32px;
+  border-radius: ${BORDERS.borderRadius16};
+  background-color: ${COLORS.white};
+`
+
+const AccordionButton = styled.button<{ isOpen: boolean }>`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
```
Comment (connected-znaim) [inline opentrons-ai-client/src/molecules/Accordion/index.tsx]: Should we be using ${ALIGN_CENTER} here?

File: opentrons-ai-client/src/molecules/Accordion/index.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  width: 100%;
+  height: auto;
+  padding: 24px 32px;
+  border-radius: ${BORDERS.borderRadius16};
+  background-color: ${COLORS.white};
+`
+
+const AccordionButton = styled.button<{ isOpen: boolean }>`
+  display: flex;
+  justify-content: space-between;
```
Comment (connected-znaim) [inline opentrons-ai-client/src/molecules/Accordion/index.tsx]: Should we use ${JUSTIFY_SPACE_BETWEEN}?

File: opentrons-ai-client/src/molecules/Accordion/index.tsx (hunk 1 above)
Comment (fbelginetw) [reply opentrons-ai-client/src/molecules/Accordion/index.tsx]: Updated, thanks!

File: opentrons-ai-client/src/molecules/Accordion/index.tsx (hunk 2 above)
Comment (fbelginetw) [reply opentrons-ai-client/src/molecules/Accordion/index.tsx]: Updated, thanks!

File: opentrons-ai-client/src/molecules/Accordion/index.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      disabled={disabled}
+    >
+      <AccordionButton
+        id={`${id}-${BUTTON}`}
+        aria-expanded={isOpen}
+        aria-controls={`${id}-${CONTENT}`}
+        isOpen={isOpen}
+        onClick={handleButtonClick}
+        aria-disabled={!isCompleted}
+        disabled={disabled}
```
Comment (shlokamin) [inline opentrons-ai-client/src/molecules/Accordion/index.tsx]: how come the values of `aria-disabled` and `disabled` are different here?

**Review by shlokamin — APPROVED** [review]
question about why `aria-disabled` and `disabled` get assigned different values but nothing blocking!

File: opentrons-ai-client/src/molecules/Accordion/index.tsx (hunk 3 above)
Comment (fbelginetw) [reply opentrons-ai-client/src/molecules/Accordion/index.tsx]: We actually don't need aria-disabled, I added the disabled property later and forgot to remove. Good catch, updated!

**Pull request merged**